### PR TITLE
feat: lib: Serde support for QString

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
See also: #943

### CODE REVIEW

File: crates/cxx-qt-lib/Cargo.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ http = { version = "1.0", optional = true }
 rgb = { version = "0.8", optional = true }
 time = { version = "0.3.20", optional = true }
 url = { version = "2.3", optional = true }
+serde = { version = "1", features=["default", "derive"], optional = true }
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/Cargo.toml:25]: Why do you need "default" ?

The docs suggest that this should work `serde = { version = "1.0", features = ["derive"] }`

https://serde.rs/derive.html

File: examples/qml_features/rust/Cargo.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ crate-type = ["staticlib"]
 [dependencies]
 cxx.workspace = true
 cxx-qt.workspace = true
-cxx-qt-lib.workspace = true
+cxx-qt-lib = { workspace = true, features=["default", "serde"] }
```
Comment (ahayzen-kdab) [inline examples/qml_features/rust/Cargo.toml:19]: same here, do we need default?

File: crates/cxx-qt-lib/Cargo.toml (hunk 1 above)
Comment (LeonMatthesKDAB) [reply crates/cxx-qt-lib/Cargo.toml]: It seems we don't, done :)

File: examples/qml_features/rust/Cargo.toml (hunk 2 above)
Comment (LeonMatthesKDAB) [reply examples/qml_features/rust/Cargo.toml]: Done

**Comment by LeonMatthesKDAB** [comment]
Let's also see what @Montel thinks of this :)

**Comment by Montel** [comment]
It's a good idea ! I like it!

**Review by Montel — APPROVED** [review]

**Pull request merged**